src/components/legend/attributes.js

```diff
@@ -74,6 +74,12 @@ module.exports = {
             'Sets the amount of vertical space (in px) between legend groups.'
         ].join(' ')
     },
+    legendtextwidth: {
+        valType: 'number',
+        min: 0,
+        editType: 'legend',
+        description: 'Sets the width (in px) of the legend.',
```

Comment:
Since we're already in the `legend` container, this one can just be called `textwidth`.

But there's a bit more to it than this that we need to discuss. The sponsor was interested in being able to specify this width as either pixels or a fraction of the plot area - similar to how colorbars can specify [thickness and thicknessmode](https://github.com/plotly/plotly.js/blob/e0a5f13a4c9b8e2f74204499168d12b2d0bab285/src/components/colorbar/attributes.js#L16-L35)

The challenge with that though is that fractional sizing should take into account the entire item - including the symbol (which already has pixel-sized `itemwidth`) and any gaps - so that you could say "make each item width 1/3 of the plot area" and the result is a legend exactly the width of the plot area, with three items on each row.

So what if the new attributes we add are:
- `legend.entrywidth` - full width of a legend entry, including symbol and padding
- `legend.widthmode` - either `'fraction'` or `'pixels'` - what `entrywidth` means (but will NOT affect `itemwidth`, would be weird for that to change with the overall plot width)
- `trace.legendwidth` - overrides `legend.entrywidth` for this trace
- `trace.legendwidthmode` - (do we need this?) overrides `legend.widthmode` for this trace